--- configure.ac	2015-01-08 02:38:43.471131639 +0000
+++ /home/xreaper/src/rubenk-pdns/configure.ac	2015-01-08 02:34:39.313131639 +0000
@@ -18,7 +18,7 @@ AC_CANONICAL_HOST
 
 AC_PROG_CC
 AM_PROG_CC_C_O
-AC_PROG_YACC
+PDNS_CHECK_BISON
 AM_PROG_LEX
 AC_PROG_INSTALL
 AC_PROG_MAKE_SET
@@ -39,13 +39,9 @@ AC_CC_STACK_PROTECTOR
 AC_CC_PARAM_SSP_BUFFER_SIZE([4])
 AC_CC_D_FORTIFY_SOURCE
 
-AC_CHECK_FUNC([socket], [], [
- AC_CHECK_LIB([socket], [socket], [LIBS="-lsocket $LIBS"])
-])
-
-AC_CHECK_FUNC([gethostent], [], [
- AC_CHECK_LIB([nsl], [gethostent], [LIBS="-lnsl $LIBS"])
-])
+AC_SEARCH_LIBS([socket], [socket])
+AC_SEARCH_LIBS([gethostent], [nsl])
+AC_SEARCH_LIBS([inet_aton], [resolv])
 
 m4_ifndef([LT_INIT],[
 AC_PROG_LIBTOOL([disable-static])
@@ -78,13 +74,13 @@ AC_CHECK_HEADERS(
 )
 
 PDNS_CHECK_RAGEL
-AC_CHECK_PROG([ASCIIDOC], [asciidoc], [asciidoc])
 
 PDNS_WITH_LUA
 
-AX_BOOST_BASE([1.35], [], [AC_MSG_ERROR([Boost not found])])
-AX_BOOST_PROGRAM_OPTIONS
-AX_BOOST_SERIALIZATION
+BOOST_REQUIRE([1.35])
+BOOST_FOREACH
+BOOST_PROGRAM_OPTIONS([mt])
+BOOST_SERIALIZATION([mt])
 
 PDNS_ENABLE_UNIT_TESTS
 
@@ -101,6 +97,9 @@ AS_IF([test "x$with_sqlite3" != "xno"],
   needsqlite3=yes
 ])
 
+PDNS_CHECK_PANDOC
+PDNS_CHECK_MKDOCS
+
 dnl Checks for typedefs, structures, and compiler characteristics.
 AC_TYPE_SIZE_T
 
@@ -140,24 +139,22 @@ AC_SUBST(LIBCRYPT)
 
 THREADFLAGS=""
 
-AM_CONDITIONAL([OS_MACOSX], false)
 case "$host_os" in
 solaris2.10)
-  LIBS="-lposix4 -lresolv -lnsl -lsocket -lpthread -lrt $LIBS"
+  LIBS="-lposix4 -lpthread -lrt $LIBS"
   CXXFLAGS="-D_REENTRANT $CXXFLAGS"
   ;;
 solaris2.8 | solaris2.9 )
   AC_DEFINE(NEED_POSIX_TYPEDEF,,[If POSIX typedefs need to be defined])
   AC_DEFINE(NEED_INET_NTOP_PROTO,,[If your OS is so broken that it needs an additional prototype])
-  LIBS="-lposix4 -lresolv -lnsl -lsocket -lpthread $LIBS"
+  LIBS="-lposix4 -lpthread $LIBS"
   CXXFLAGS="-D_REENTRANT $CXXFLAGS"
   ;;
 linux*)
   THREADFLAGS="-pthread"
   ;;
 darwin*)
-  CXXFLAGS="-D__APPLE_USE_RFC_3542 $CXXFLAGS"
-  AM_CONDITIONAL([OS_MACOSX], true)
+  CXXFLAGS="-D__APPLE_USE_RFC_3542 -D_XOPEN_SOURCE $CXXFLAGS"
   ;;
 *)
   LDFLAGS="-pthread $LDFLAGS"
@@ -299,17 +296,6 @@ for a in $modules; do
 
   if test ${a} = "gpgsql"; then
     LIBS="$LIBS $LIBCRYPT"
-
-    case "$host_os" in
-      freebsd*)
-        ;;
-      darwin*)
-        modulelibs="$modulelibs -lresolv"
-        ;;
-      *)
-        modulelibs="$modulelibs -lresolv -lnsl"
-        ;;
-    esac
   fi
 done
 
@@ -319,7 +305,7 @@ done
 
 CFLAGS="$PIE_CFLAGS $CFLAGS"
 CXXFLAGS="$PIE_CFLAGS $CXXFLAGS"
-LDFLAGS="$RELRO_LDFLAGS $BOOST_LDFLAGS $LDFLAGS"
+LDFLAGS="$RELRO_LDFLAGS $LDFLAGS"
 
 AC_SUBST(LIBS)
 
@@ -330,11 +316,14 @@ AC_SUBST([AM_CPPFLAGS],
 AC_SUBST([YAHTTP_CFLAGS], ['-I$(top_srcdir)/pdns/ext/yahttp'])
 AC_SUBST([YAHTTP_LIBS], ['-L$(top_builddir)/pdns/ext/yahttp/yahttp -lyahttp'])
 
+export moduledirs moduleobjects modulelibs
+
 AC_CONFIG_FILES([
   Makefile
   modules/Makefile
   pdns/Makefile
   codedocs/Makefile
+  docs/Makefile
   pdns/pdns.init
   pdns/ext/Makefile
   pdns/ext/yahttp/Makefile
@@ -362,3 +351,23 @@ AC_CONFIG_FILES([
 ])
 AC_OUTPUT
 
+AC_MSG_NOTICE([])
+AC_MSG_NOTICE([Configuration summary])
+AC_MSG_NOTICE([=====================])
+AC_MSG_NOTICE([])
+AC_MSG_NOTICE([Modules: $modules])
+AC_MSG_NOTICE([Dynamic Modules: $dynmodules])
+AC_MSG_NOTICE([])
+AC_MSG_NOTICE([CC: $CC])
+AC_MSG_NOTICE([CXX: $CXX])
+AC_MSG_NOTICE([LD: $LD])
+AC_MSG_NOTICE([CFLAGS: $CFLAGS])
+AC_MSG_NOTICE([CPPFLAGS: $CPPFLAGS])
+AC_MSG_NOTICE([CXXFLAGS: $CXXFLAGS])
+AC_MSG_NOTICE([LDFLAGS: $LDFLAGS])
+AC_MSG_NOTICE([])
+AC_MSG_NOTICE([BOOST_CPPFLAGS: $BOOST_CPPFLAGS])
+AC_MSG_NOTICE([])
+AC_MSG_NOTICE([SQLite3 support: $with_sqlite3])
+AC_MSG_NOTICE([Lua support: $with_lua])
+AC_MSG_NOTICE([])
